lib/generators/datepicker/datepicker_generator.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails/generators/base'
+
+class DatepickerGenerator < Rails::Generators::Base
+
+  source_root File.expand_path('../templates', __FILE__)
+
+  def modify_application_js
+    js_file
+    assets = ['datepicker', 'jquery.mask']
+    assets.each do |asset|
+      if File.open(js_root_path + js_file).grep(/(\/\/= require #{asset})/).size > 0
+        puts "//= require #{asset} already exists in #{js_file}..."
+      else
+        open(js_root_path + js_file, 'a') do |file|
+          file << "//= require #{asset}\n"
+        end
+        puts "//= require #{asset} added to #{js_file}..."
+      end
+    end
+  end
+
+  def modify_application_css
+    style_file
+    text = File.read(style_root_path + style_file) unless style_file.nil?
+    if File.readlines(style_root_path + style_file).grep(/(\*\= require datepicker)/).size > 0
```

Comment:
Some idea here.  A method like `datepicker_already_required_in_style_manifest?` will make the `modify_application_css`  method more readable.  It also removes complexity by offloading some of the "weight" of `modify_application_css` to another method.  It also sets up up better for "reuse" in the future.